From 0380883e47e098192203583a65d1644371d03740 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Sat, 19 Jun 2021 15:48:29 +0530
Subject: [PATCH 2/7] Revert "octeontx2-af: Disable packet TX and RX"

commit 5358d8522d1993e9581db588a153fbfe4e3c0180 from
git@git.assembla.com:cavium/WindRiver.linux.git

CGM LMAC packet TX and RX are disabled after
bringing up LMAC ports during AF driver
initialization. This was done because packets
were received by CGX and till NPC though
interface is not up and running. LMAC's packet
TX and RX are only enabled in the interface
bring up sequence. But there exists a race condition
when an interface is brought up early while kernel
booting (like NFS boot or setting ip kernel
commandline parameter) so that packet TX and RX
are enabled and at later stage LMAC port
up work handler is invoked and disables TX and RX.
Hence after kernel boots interface ends up
in a case where interface shows <UP, RUNNING>
but packet I/O does not work. This patch reverts
the commit "ba9834332a37" to fix the problem.
Reverting this will cause rx_miss_stats(drops at NPC)
to increment though no interface is up.

Change-Id: If01353f50b13b2539dc5dac9df2a34104e62f613
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/54522
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/cgx.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
index 7aec416c2bc5..5dd86bf77a11 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
@@ -1487,10 +1487,6 @@ static void cgx_lmac_linkup_work(struct work_struct *work)
 		if (err)
 			dev_info(dev, "cgx port %d:%d Link up command failed\n",
 				 cgx->cgx_id, i);
-		/* Disable packet I/O in CGX in case firmware enabled it
-		 * after Link up sequence.
-		 */
-		cgx_lmac_rx_tx_enable(cgx, i, false);
 	}
 }
 
@@ -1592,8 +1588,6 @@ static int cgx_lmac_init(struct cgx *cgx)
 		/* Add reference */
 		cgx->lmac_idmap[i] = lmac;
 		cgx_lmac_pause_frm_config(cgx, i, true);
-		/* Disable packet I/O for a sane state */
-		cgx_lmac_rx_tx_enable(cgx, lmac->lmac_id, false);
 	}
 
 	return cgx_lmac_verify_fwi_version(cgx);
-- 
2.31.1

